fearless/Common/DataProvider/Sources/Rewards/SubqueryTransfersSource.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import RobinHood
+import FearlessUtils
+import BigInt
+
+struct SubqueryTransferData: Decodable {
+    let transferId: String
+    let toId: String
+    let fromId: String
+    let amount: Decimal?
+    let createdAt: String
+
+    init?(from json: JSON, chain: Chain) {
+        guard
+            let transferId = json.id?.stringValue,
+            let fromId = json.fromId?.stringValue,
+            let toId = json.toId?.stringValue,
+            let createdAt = json.createdAt?.stringValue
+        else { return nil }
+
+        let amount = json.amount?.stringValue
+            .map { BigUInt($0) }?
+            .map { Decimal.fromSubstrateAmount($0, precision: chain.addressType.precision) }
+            ?? Decimal(0)
+
+        self.transferId = transferId
+        self.toId = toId
+        self.fromId = fromId
+        self.createdAt = createdAt
+        self.amount = amount
+    }
+}
+
+final class SubqueryTransfersSource {
+    typealias Model = [SubqueryTransferData]
+
+    let address: String
+    let url: URL
+    let chain: Chain
+
+    init(
+        address: String,
+        url: URL,
+        chain: Chain
+    ) {
+        self.address = address
+        self.url = url
+        self.chain = chain
+    }
+}
+
+extension SubqueryTransfersSource: SingleValueProviderSourceProtocol {
```

Comment:
I think we should use pagination here. So consider to use DataProviderSourceProtocol and DataProvider instead of SingleValueProvider